### PR TITLE
HPCC-16249 Couchbaseembedded - All string results were inexplicably nulled out

### PR DESCRIPTION
**Pull request by rpastrana**
Signed-off-by: rpastrana rodrigo.pastrana@lexisnexis.com

### CODE REVIEW

**Comment by rpastrana** [comment]
@RussWhitehead please review.

File: plugins/couchbase/couchbaseembed.cpp [hunk 1]
```diff
@@ -512,9 +510,11 @@ namespace couchbaseembed
             unsigned numchars = rtlUtf8Length(strlen(value), value);
             rtlUtf8ToStrX(chars, result, numchars, value);
         }
-
-        NullFieldProcessor p(NULL);
-        rtlStrToStrX(chars, result, p.resultChars, p.stringResult);
+        else
+        {
+            NullFieldProcessor p(NULL);
```
Comment (RussWhitehead) [inline plugins/couchbase/couchbaseembed.cpp]: Would it make sense to make NullFieldProcessor a class member since it seems to be used a lot?

**Comment by RussWhitehead** [comment]
@rpastrana Looks good with one minor question

**Comment by rpastrana** [comment]
@RussWhitehead please review.

**Comment by richardkchapman** [comment]
The changelog would be more meaningful if the commit title referred to couchbase

**Comment by RussWhitehead** [comment]
Possibly beyond this scope of this PR, but I see at least 6 instances of local NullFieldProcessor being constructed that would benefit from using your new static global one. Otherwise looks good

**Comment by rpastrana** [comment]
@RussWhitehead the few instances of NullFieldProcessor left take in a field object. the others took a NULL and we could use the static const member.

**Comment by rpastrana** [comment]
@richardkchapman title updated, please merge

**Pull request merged**